tests/framework/utils.py

```diff
@@ -563,25 +563,21 @@ def compare_versions(first, second):
     return first[0] - second[0]
 
 
-class ResultsFileDumper:
-    """Class responsible with outputting test results to files."""
+def performance_tests_cleanup(basevm, file_dumper, result=None, fail_err=None):
```

Comment:
hmm I'm thinking that it would be even easier to supply the `results_file_dumper` to the statistic core/run_exercise function. This would offload the responsibility to handle the exception and dumping the results from the test.

Also, let's make sure we add a mechanism for killing the VMs after in this PR. The destructor way sounds good to me